Remove pyyaml dependency in tests and update instructions

aburgm opened the pull request:
Without `pyyaml`, the following error occurs when running tests:

```
=================================== ERRORS ====================================
________________ ERROR collecting tests/conda_env/test_env.py _________________
tests\conda_env\test_env.py:7: in <module>
    import yaml
E   ImportError: No module named yaml
________________ ERROR collecting tests/conda_env/test_env.py _________________
ImportError while importing test module 'C:\Users\traveler\AppData\Local\conda\conda\tests\conda_env\test_env.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests\conda_env\test_env.py:7: in <module>
    import yaml
E   ImportError: No module named yaml
!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!
```

kalefranz commented:
The instructions do need updated.  In the 4.3 and up branches, it would probably be something like

```
pip install -r utils requirements-test.txt
python utils/setup-testing.py develop
```

We're not using `pyyaml` anymore though.  We're usual `ruamel_yaml`.  Rather than mislead with the `pyyaml` instruction, I'd rather just fix the test file `tests/conda_env/test_env.py` by replacing the `import yaml` on line 7 with

```python
from conda.common.yaml import get_yaml
yaml = get_yaml()
```

aburgm commented:
Thanks, I've updated the instructions and fixed the conda_env test.

The previous instructions seemed to work fine for me, though. Are the new instructions so that they can also run from a virtualenv instead of a miniconda install?

While at it, would you also suggest to replace the instructions for setting up a dev build with installing `requirements.txt` as well?

kalefranz commented:
targeting 4.4.x via #5137
